### PR TITLE
Fix connection leak in isotope

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/16735

Basically if there was an error (5xx), then we never closed the request
body, leading to a leak of a connection.

I also had to update the dockerfile and add go.mod

### CODE REVIEW

**Comment by carolynhu** [comment]
Why do you delete some of those licenses file? like fortio. Don't we need them?

File: perf/load/values.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-serviceGraphImage: tahler/isotope-service:1
+serviceGraphImage: gcr.io/istio-testing/isotope:0.0.1
```
Comment (carolynhu) [inline perf/load/values.yaml:1]: Who is responsible for building and pushing isotope image? Also the versioning.

**Comment by howardjohn** [comment]
/hold

this is breaking something, requests are not getting sent properly for some reason. Will investigate and update tomorrow

File: perf/load/values.yaml (hunk 1 above)
Comment (howardjohn) [reply perf/load/values.yaml:1]: We could set up the full CI/CD but this image has changed 1 time in the past 1.5years+, so I am not sure its needed?

**Comment by howardjohn** [comment]
Found the issue, the probability code was wrong. I am not sure how it worked before, possibly when we updated deps it was broken? I also switched the logging to istio log to remove the dependency on fortio

**Comment by howardjohn** [comment]
> Why do you delete some of those licenses file? like fortio. Don't we need them?

This is autogenerated based on what is used in the code. We no longer are importing fortio

File: isotope/service/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+COPY . /build/
 
-WORKDIR /go/src/istio.io/tools/isotope/service
+WORKDIR /build
 
-RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux \
-    go build -a -installsuffix cgo -o ./main ./main.go
+RUN GOARCH=amd64 CGO_ENABLED=0 GOOS=linux go build -o isotope_service ./service
+
+RUN ls -la
```
Comment (sdake) [inline isotope/service/Dockerfile:14]: debug output

File: isotope/service/pkg/srv/executable.go [hunk 3]
```diff
@@ -25,13 +25,16 @@ import (
...
-	"fortio.org/fortio/log"
-
+	"istio.io/pkg/log"
 	"istio.io/tools/isotope/convert/pkg/graph/script"
 	"istio.io/tools/isotope/convert/pkg/graph/svctype"
 	"istio.io/tools/isotope/service/pkg/srv/prometheus"
 )
 
+func init() {
+	rand.Seed(time.Now().UnixNano())
```
Comment (sdake) [inline isotope/service/pkg/srv/executable.go:35]: If you do a rev of this PR, you may consider adding a comment here that random number usage in this code is not for a cryptographic purpose. rand.Seed() (may or should) trigger some various security linters. These annoying linters don't add much value, but are required by various CISO (chief information security officer) peeps for no good reason. There may be multiple CISOs in one firm...

**Comment by howardjohn** [comment]
Updated per comments, PTAL

**Comment by howardjohn** [comment]
@sdake please take a look. Thanks!

**Comment by howardjohn** [comment]
@sdake please take a look. Thanks!

**Comment by mandarjog** [comment]
Thanks John this is awesome! This issue was open for such a long time !

**Pull request merged**